todolist/parser.go

```diff
@@ -28,6 +30,25 @@ func (p *Parser) ParseNewTodo(input string) *Todo {
 	return todo
 }
 
+// Parse accepts user input and splits it into subcommand, the todo id to
+// work on and the input to the subcommand function.
+func (p Parser) Parse() (subcommand string, id int, input string) {
+	r := regexp.MustCompile(`(\w+)\s+(\d+)\s+(.*)`)
+	matches := r.FindStringSubmatch(p.input)
+	if len(matches) < 4 {
+		fmt.Println("Could match command, id or subject")
+		return "", -1, ""
+	}
+
+	// because of the regexp match, this can never fail
```

Comment:
no need for this comment. it's easy to see what will happen in the next few lines of code.